### PR TITLE
2019 profile resources module

### PR DESCRIPTION
**Pull request by bganapa**
## Description

  Azure stack supports 2018-05-01 api version with 2019-03-01-hybrid profile

    Copy of the AzureRM.Resources module 6.4.2, commit hash 9d2504d5012994fd43860c5c114b4e7fc11a04b9

    PolicyDefinition(2016-12-01) and PolicyAssisgment(2017-06-01-preview) operations are still with old api version

    SDK used 1.9.1-preview

    Managementgroups cmdlets included though azurestack does not support it

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by vladimir-shcherbakov** [comment]
@knithinc  @viananth  Could you please  review the PR?

File: src/StackAdmin/Resources/Commands.Resources/AzureRM.Resources.psd1 [hunk 1]
```diff
@@ -74,62 +75,70 @@ NestedModules = @('.\Microsoft.Azure.Commands.Resources.dll',
...
+               'Set-AzureRmResource', 'New-AzureRmResource', 
+               'Set-AzureRmResourceLock', 'Get-AzureRmResource', 
+               'Get-AzureRmManagedApplicationDefinition', 
+               'New-AzureRmManagedApplicationDefinition', 
+               'Set-AzureRmManagedApplicationDefinition', 
+               'Remove-AzureRmManagedApplicationDefinition', 
+               'Get-AzureRmManagedApplication', 'New-AzureRmManagedApplication', 
+               'Set-AzureRmManagedApplication', 'Remove-AzureRmManagedApplication', 
+               'Get-AzureRmManagementGroup', 'New-AzureRmManagementGroup', 
+               'Update-AzureRmManagementGroup', 'Remove-AzureRmManagementGroup',
```
Comment (bganapa) [inline src/StackAdmin/Resources/Commands.Resources/AzureRM.Resources.psd1]: Remove the management group related cmdlets from the exported cmdlet list as they are not supported in AzureStack

File: src/StackAdmin/Resources/Commands.Resources/Commands.Resources.csproj [hunk 2]
```diff
@@ -49,8 +49,11 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="Microsoft.Azure.Management.ManagementGroups, Version=1.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.ManagementGroups.1.1.1-preview\lib\net452\Microsoft.Azure.Management.ManagementGroups.dll</HintPath>
```
Comment (bganapa) [inline src/StackAdmin/Resources/Commands.Resources/Commands.Resources.csproj]: replace with $(LibraryNugetPackageFolder)

File: src/StackAdmin/Resources/Commands.Resources.Test/ScenarioTests/ResourcesController.cs [hunk 3]
```diff
@@ -53,20 +57,21 @@ public sealed class ResourcesController
 
         public ResourceManagementClient ResourceManagementClient { get; private set; }
 
-        public LegacyRMClient.ResourceManagementClient LegacyResourceManagementClient { get; private set; }
+        //public LegacyRMClient.ResourceManagementClient LegacyResourceManagementClient { get; private set; }
```
Comment (bganapa) [inline src/StackAdmin/Resources/Commands.Resources.Test/ScenarioTests/ResourcesController.cs]: REmove the commented out code

File: src/StackAdmin/Resources/Commands.Resources.Test/ScenarioTests/ResourcesController.cs [hunk 4]
```diff
@@ -171,33 +191,92 @@ public void RunPsTest(params string[] scripts)
...
+                {
+                    cleanup?.Invoke();
+                }
+            }
+        }
         private void SetupManagementClients(MockContext context)
         {
-            LegacyResourceManagementClient = GetLegacyResourceManagementClient();
-            LegacySubscriptionClient = GetLegacySubscriptionClient();
+            //LegacyResourceManagementClient = GetLegacyResourceManagementClient();
```
Comment (bganapa) [inline src/StackAdmin/Resources/Commands.Resources.Test/ScenarioTests/ResourcesController.cs]: Remove the commented codeblock

File: src/StackAdmin/Resources/Commands.ResourceManager/Cmdlets/Entities/Resources/ResourceIdentityType.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Entities.Resources
+{
+    /// <summary>
+    /// The type of resource identity that is assigned to a resource.
+    /// </summary>
+    public class ResourceIdentityType
```
[the remark this answers is not in the document]
Comment (bganapa) [reply src/StackAdmin/Resources/Commands.ResourceManager/Cmdlets/Entities/Resources/ResourceIdentityType.cs]: I checked and this is not exposed as a separate parameter. This is exposed via Properties and it is optional. It wont cause any issue, though we dont support it

**Comment by MiYanni** [comment]
@bganapa Closing. Likely, this aspect will be handed by the generator. So, we're currently not looking at enhancing/updating AzureRM for Stack.

File: src/StackAdmin/Resources/Commands.ResourceManager/Cmdlets/Components/ApiVersionHelper.cs [hunk 6]
```diff
@@ -146,11 +145,10 @@ private class ApiVersionCache
 
             static ApiVersionCache()
             {
-// TODO: Remove IfDef
-#if NETSTANDARD
-                _cache = new MemoryCache(new MemoryCacheOptions());
-#else
+#if !NETSTANDARD
```
Comment (markcowl) [inline src/StackAdmin/Resources/Commands.ResourceManager/Cmdlets/Components/ApiVersionHelper.cs]: We can remove these - these are never compiled as NETSTANDARD

Comment (bganapa) [reply src/StackAdmin/Resources/Commands.ResourceManager/Cmdlets/Components/ApiVersionHelper.cs]: fixed them

**Comment by maddieclayton** [comment]
@bganapa Is this ready for review and merge?

**Comment by bganapa** [comment]
@maddieclayton  Thanks for the ping. This is ready for review and merge

File: src/StackAdmin/Resources/Commands.ResourceManager/Cmdlets/Properties/AssemblyInfo.cs [hunk 7]
```diff
@@ -25,8 +25,8 @@
 [assembly: ComVisible(false)]
 [assembly: CLSCompliant(false)]
 [assembly: Guid("e8f34267-c461-4eae-b156-5f3528553d10")]
-[assembly: AssemblyVersion("6.0.2")]
-[assembly: AssemblyFileVersion("6.0.2")]
+[assembly: AssemblyVersion("6.1.0")]
```
Comment (maddieclayton) [inline src/StackAdmin/Resources/Commands.ResourceManager/Cmdlets/Properties/AssemblyInfo.cs]: This doesn't match the version in the psd1

**Pull request merged**